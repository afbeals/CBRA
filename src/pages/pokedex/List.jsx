// Extenal
import React, { useState } from 'react';
import PropTypes from 'prop-types';

// Local
import withPokelist from '~Modules/pokelist/withPokelist';
import withPokedex from '~Modules/pokedex/withPokedex';
import ErrorBoundary from '~Components/ErrorBoundary';
import Filter from './list/Filter';

// Constants
const classname = 'list';

// Component
const List = ({
  pClassname,
  selectPokemon,
  getPokelist,
  getSelectedPokelist: regionPokeList,
  getIsFetchingPokemon: isFetching,
}) => {
  const [filteredList, updateFiltered] = useState(null);
  const handleUpdateFiltered = newList => {
    updateFiltered(newList);
  };
  const pokeList = filteredList || getPokelist;
  return (
    <ErrorBoundary>
      <div data-testid="pokedex-list" className={`${pClassname}__${classname}`}>
        <Filter
          pClassname={`${pClassname}__${classname}`}
          updateFiltered={handleUpdateFiltered}
          orgList={getPokelist}
          isLoading={isFetching}
        />
        <ul className={`${pClassname}__${classname}__pokemon`}>
          {!!getPokelist &&
            regionPokeList.map(pokemon => {
              if (!pokeList[pokemon]) return null;
              const pId = pokemon;
              const { name } = pokeList[pokemon];
              return (
                <li
                  key={`pokemonId-${pokemon}`}
                  className={`${pClassname}__${classname}__pokemon__item`}
                >
                  <button
                    type={'button'}
                    onClick={() => selectPokemon(pId)}
                    onKeyDown={() => selectPokemon(pId)}
                  >
                    <div className={'name'}>
                      <p className={'name__pre'}>name:</p>
                      <p className={'name__data'}>{name}</p>
                    </div>
                    <div className={'pId'}>
                      <p className={'pId__pre'}>No:</p>
                      <p className={'pId__data'}>{pId}</p>
                    </div>
                  </button>
                </li>
              );
            })}
        </ul>
      </div>
    </ErrorBoundary>
  );
};

List.propTypes = {
  pClassname: PropTypes.string.isRequired,
  selectPokemon: PropTypes.func.isRequired,
  getIsFetchingPokemon: PropTypes.bool.isRequired,
  getPokelist: PropTypes.object,
  getSelectedPokelist: PropTypes.array,
};

List.defaultProps = {
  getSelectedPokelist: [],
};

export default withPokedex(withPokelist(List));

export { List };
