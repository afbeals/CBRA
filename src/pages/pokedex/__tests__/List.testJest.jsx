// External
import React from 'react';
import { render, cleanup } from '@testing-library/react';

// Local
import { List } from '../List';

// tests
describe('List', () => {
  afterEach(cleanup);
  let fnClick = null;
  beforeEach(() => {
    fnClick = jest.fn();
  });
  it('should match List to snapshot', () => {
    const { getByTestId } = render(
      <List
        pClassname={'parent'}
        selectPokemon={fnClick}
        getIsFetchingPokemon={false}
      />,
    );
    const list = getByTestId('pokedex-list');
    expect(list).toMatchSnapshot();
  });

  it('should render the pokemon list', () => {
    const { getByText } = render(
      <List
        pClassname={'parent'}
        selectPokemon={fnClick}
        getIsFetchingPokemon={false}
        getPokelist={{
          1: {
            name: 'bulbasaur',
            url: '/',
            urlId: 1,
          },
        }}
        getSelectedPokelist={[1]}
      />,
    );
    const list = getByText('bulbasaur');
    expect(list).toBeTruthy();
  });
});
