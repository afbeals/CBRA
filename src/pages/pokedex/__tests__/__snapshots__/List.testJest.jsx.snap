// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match List to snapshot 1`] = `
<div
  class="parent__list"
  data-testid="pokedex-list"
>
  <div
    class="parent__list__filters"
    data-testid="list_filter"
  >
    <div
      class="parent__list__filters__decor"
    >
      <div
        class="parent__list__filters__decor__light"
      >
        <div
          class="trim"
        >
          <div
            class="bulb"
          />
        </div>
      </div>
      <div
        class="parent__list__filters__decor__sound"
      >
        <div
          class="bar"
        />
        <div
          class="bar"
        />
        <div
          class="bar"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root textField MuiFormControl-marginDense"
      data-testid="textfield"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-filled"
        data-shrink="false"
      >
        Pokemon Name...
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <ul
    class="parent__list__pokemon"
  />
</div>
`;
